### ISSUE TITLE
SPI size limit

### ISSUE THREAD

**Issue by empirephoenix** [Pi4J/pi4j-v1#197]
Hi,

is there a reason for the hardcoded limit of the buffer in the SPI..c class for the buffer?
https://github.com/Pi4J/pi4j/blob/master/pi4j-native/src/main/native/com_pi4j_wiringpi_Spi.c
In my opinion it should be possible to instead determine the necessary size with the information from the java side? 

```
   char buffer[2048];
   int len = (*env)->GetStringUTFLength(env, data);
   (*env)->GetStringUTFRegion(env, data, 0, len, buffer);
```

instead of 

```
   int len = (*env)->GetStringUTFLength(env, data);
   char buffer[len]; //kinda, might be wrong length for this specific type, but you get the idea
   (*env)->GetStringUTFRegion(env, data, 0, len, buffer);
```

So the question is, are there any good reason, a fixed buffer is used instead? 
If not I would create a pull request for this, allowing variable sized data.

**Comment by natdan** [comment]
I suspect that it was just least resistance path (at the point it was coded). Determining buffer size from Java side makes more sense. I think you should create pull request anyway and if Robert is OK with that we'll pull it in.

**Comment by savageautomate** [comment]
Hi empirephoenix,

This may be a duplicate of issue: https://github.com/Pi4J/pi4j/issues/143
This has been fixed in the 'develop' branch and released in the 1.1-SNAPSHOT builds.  Please try the newer build to see if that fixes the problem for you.

Thank You,
Robert

**Comment by empirephoenix** [comment]
Yes, you are right, it is already done in the develop. Thanks :+1:
